packages/cozy-client/src/flagship-certification/flagship-certification.spec.js

```diff
@@ -0,0 +1,235 @@
+import { certifyFlagship } from './flagship-certification'
+import { createMockClient } from '../mock'
+
+import { getAppAttestationFromStore } from './store-attestation'
+
+jest.mock('./store-attestation', () => ({
+  getAppAttestationFromStore: jest.fn()
+}))
+
+const getClientMock = () => {
+  const client = createMockClient({
+    clientOptions: {
+      oauth: {}
+    }
+  })
+
+  client.stackClient.oauthOptions = {
+    clientID: 'SOME_CLIENT_ID',
+    registrationAccessToken: 'SOME_REGISTRATION_ACCESS_TOKEN'
+  }
+
+  return client
+}
+
+const mockCorrectChallengeRequest = client => {
+  client.stackClient.fetchJSON.mockResolvedValueOnce({
+    nonce: 'SOME_NONCE'
+  })
+}
+
+const mockCorrectStoreApiRequest = () => {
+  getAppAttestationFromStore.mockResolvedValueOnce({
+    platform: 'android',
+    attestation: 'SOME_STORE_ATTESTATION',
+    keyId: 'SOME_KEY_ID'
+  })
+}
+
+const mockCorrectCertificationConfig = () => {
+  return {
+    androidSafetyNetApiKey: 'SOME_ANDROID_SAFETY_NET_API_KEY'
+  }
+}
+
+describe('certifyFlagship', () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'warn').mockImplementation(() => jest.fn())
+  })
+
+  afterEach(() => {
+    jest.restoreAllMocks()
+    getAppAttestationFromStore.mockReset()
+  })
+
+  it('should correctly call certification endpoints', async () => {
+    const client = getClientMock()
+
+    const certificationConfig = mockCorrectCertificationConfig()
+
+    mockCorrectChallengeRequest(client)
+
+    mockCorrectStoreApiRequest()
+
+    await certifyFlagship(certificationConfig, client)
+
+    // Should ask the stack for a challenge
```

Comment:
instead of using comment in  the test, let's split the test =>
```
it('should ask the stack for a challenge', () => {
// this expect below only
```

Reply:
same for all other comments

Reply:
I'm not sure, I want to test this specific call combination as the certification process require to ask a challenge to the stack, then query the Store api, then give the result to the stack. Multiple test cases would not allow to test the sequential order, would it?

Reply:
ok, then, instead of using comment I suggest to rename the test:

'it should ask the stack for a challenge, before doing B action, then doing C, and finally D'